### ISSUE TITLE
Strange checksum errors indice during the resilvering

### ISSUE THREAD

**Issue by romanshein** [openzfs/zfs#12134]
Distribution Name	: Proxmox
Distribution Version : 6.4
Linux Kernel	| 5.4.114-1-pve 
Architecture	| x64
ZFS Version	| 2.0.4-pve1
SPL Version	| 2.0.4-pve1

root@debian:~# zpool status z6
  pool: z6
 state: ONLINE
**status: One or more devices is currently being resilvered.  The pool will
        continue to function, possibly in a degraded state.**
action: Wait for the resilver to complete.
  scan: resilver in progress since Thu May 27 11:30:38 2021
        1.49T scanned at 676M/s, 1.14T issued at 520M/s, 3.42T total
        195G resilvered, 33.49% done, 01:16:20 to go
config:

        NAME             STATE     READ WRITE CKSUM
        z6               ONLINE       0     0     0
          raidz2-0       ONLINE       0     0     0
            sdd3         ONLINE       0     0     0
            sdf3         ONLINE       0     0     0
            sdh1         ONLINE       0     0     0
            sdi2         ONLINE       0     0     0
            replacing-4  ONLINE       0     0   **237**
              sdj1       ONLINE       0     0     0  (resilvering)
              sdl        ONLINE       0     0     0  (resilvering)
            sdk2         ONLINE       0     0     0
        special
          mirror-1       ONLINE       0     0     0
            sdg10        ONLINE       0     0     0
            sdb4         ONLINE       0     0     0
        logs
          sdb10          ONLINE       0     0     0
          sdg11          ONLINE       0     0     0
        cache
          sde7           ONLINE       0     0     0

The CKSUM errors for replacing-4 are supposed to be a sum of errors for the devices, yet it is clearly detached from the underlying disks. Meanwhile, the status line does not indicate any problems with the pool.

Upon the resilver, no signs of error. The status is 100% ok.

**Comment by GregorKopka** [comment]
Why is the stale bot still operating on issues marked as 'defect' ?

**Comment by behlendorf** [comment]
> The CKSUM errors for replacing-4 are supposed to be a sum of errors for the devices

Not exactly, when the CKSUM errors are reported against the `replacing` vdev instead of one of its children it's because ZFS couldn't definitively identify which of the children were damaged.  In this case both children will be repaired.

This behavior looks correct to me for the case where unexpected silent corruption was discoverer when resilvering.

**Comment by GregorKopka** [comment]
> This behavior looks correct to me for the case where unexpected silent corruption was discoverer when resilvering.

I would understand that CKSUM errors being detected on the drive that is being replaced with a new one, but not on the new drive that is the target of the replacement. 

Or is the new drive already being used to serve reads _while still being written to_?
_Then_ I would understand why no statistics are kept for the individual drives, as that would generate massive amounts of errors from the new drive when being accessed in any way (especially at the start of resilver), leading to admin confusion/panic.

**Comment by romanshein** [comment]
> Or is the new drive already being used to serve reads _while still being written to_?

1) Potentially, it could be the case, because I, indeed, use ZFS pools made of partitions, rarely whole disks, for flexibility. Yet I do not remember what was the case back 2 years ago.
2) From your description it sounds like CKSUM isn't really a checksumming error counter, but the counter for all kinds of disk errors. Maybe it makes sense to rename the column to something like "DEVERRORS"? This way it will provide better guidance with troubleshooting...
